event-hubs.api.md

```diff
@@ -0,0 +1,287 @@
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export interface LastEnqueuedEventInfo {
+    enqueuedTime?: Date;
+    offset?: string;
+    retrievalTime?: Date;
+    sequenceNumber?: number;
+}
+
+export { MessagingError }
+
+// @public
+export interface PartitionCheckpointer {
+    updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
+    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
+    // @internal (undocumented)
+    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
+}
+
+// @public
+export interface PartitionContext {
+    consumerGroupName: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    partitionId: string;
+}
+
+// @public
+export interface PartitionManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export interface PartitionOwnership extends PartitionContext {
+    eTag?: string;
+    lastModifiedTimeInMs?: number;
+    ownerId: string;
+    ownerLevel: number;
+}
+
+// @public
+export interface PartitionProperties {
+    beginningSequenceNumber: number;
+    eventHubName: string;
+    lastEnqueuedOffset: number;
+    lastEnqueuedSequenceNumber: number;
+    lastEnqueuedTimeUtc: Date;
+    partitionId: string;
+}
+
+// @public
+export type ProcessCloseHandler = (reason: CloseReason, context: SubscriptionPartitionContext) => Promise<void>;
+
+// @public
+export type ProcessErrorHandler = (error: Error, context: SubscriptionPartitionContext) => Promise<void>;
+
+// @public
+export type ProcessEvent = (receivedEvent: ReceivedEventData, context: PartitionContext & PartitionCheckpointer) => Promise<void>;
+
+// @public
+export type ProcessInitializeHandler = (context: SubscriptionPartitionContext & SubscriptionPartitionInitializer) => Promise<void>;
+
+// @public
+export interface ReceivedEventData {
+    body: any;
+    enqueuedTimeUtc: Date;
+    offset: number;
+    partitionKey: string | null;
+    properties?: {
+        [key: string]: any;
+    };
+    sequenceNumber: number;
+    systemProperties?: {
+        [key: string]: any;
+    };
+}
+
+export { RetryOptions }
+
+// @public
+export interface SendBatchOptions {
+    abortSignal?: AbortSignalLike;
+    parentSpan?: Span | SpanContext;
+}
+
+// @public
+export interface Subscription {
+    close(): Promise<void>;
+    isRunning: boolean;
+}
+
+// @public
+export interface SubscriptionEventHandlers {
+    processClose?: ProcessCloseHandler;
+    processError?: ProcessErrorHandler;
```

Comment:
Just wondering - I believe that the analog here in .NET is `ProcessExceptionAsync` on `EventProcessorClient`.  In .NET, the documentation states that this function **must** be implemented, yet here we do not require this.  Is the difference concerning?

Reply:
One can still use the library without providing the error handler. So forcing one to provide the error handler goes against the "lets make it easy for people to get started" theory.

Reply:
It's an interesting point though - if we want to lead the user to good behaviors then showing them this particular piece right away (even if left empty) would be a good idea.

At the very least it makes it obvious in a code review that there's no error handling.